packages/components/src/Badge/Badge.scss

```diff
@@ -20,6 +20,27 @@ $tc-badge-small-padding: 0.8rem;
 
 $tc-badge-disabled-opacity: 0.62;
 
+@mixin colored-badge($background-color, $color) {
+	.tc-badge-button {
+		background-color: $background-color;
+		color: $color;
+
+		.tc-badge-separator {
+			background-color: $color;
+		}
+
+		.tc-badge-category,
```

Comment:
Could you extract this style reset and use it once with `[class^="tc-badge--"]` or something like that:
```js
const containerClasses = theme('tc-badge', displayClass, className, {
  //...
  'tc-badge--colored': !!type,
  [`tc-badge--${type}`]: !!type,
});```